plugins/catalog-react/src/hooks/useEntityPermission.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { stringifyEntityRef } from '@backstage/catalog-model';
+import { Permission } from '@backstage/plugin-permission-common';
+import { usePermission } from '@backstage/plugin-permission-react';
+import { useEntity } from './useEntity';
+
+/**
+ * A thin wrapper around the
+ * {@link @backstage/plugin-permission-react#usePermission} hook which uses the
+ * current entity in context to make an authorization request for the given
+ * permission.
+ * @public
+ */
+export function useEntityPermission(permission: Permission) {
+  const { entity } = useEntity();
+  if (!entity) {
+    throw new Error('No entity in current context.');
```

Comment:
this will make the component crash in case the entity is loading

Reply:
Seems like this works because the context menu isn't rendered until the entity is loaded, but this hook should probably handle the loading state to avoid misleading behavior in other use cases.

Reply:
I find the type casting in the implementation of `useEntity` a bit strange because it's casting `undefined` as `T`, so the return signature just looks like `entity` will always be `Entity`, whereas in reality it should be `Entity | undefined`. But it's a good point in terms of handling the loading case. Due to the no-conditional-calls rule of hooks I've had to make the hook a bit more complicated, but it should handle all cases now.

Reply:
Oh yea that is strange! Didn't notice that before.. 🤔